### PR TITLE
Add generation and publish process for an out-of-box docker image of Hydra Lab center-agent assembly

### PR DESCRIPTION
**Pull request by olivershen-wow**
- Add a out-of-box docker for center-agent assembly
  - Updated config ymls of docker version for agent/center each.
  - Passed agent variables during docker build.
  - Blob connection string should be passed to docker run after pulling the image.
  - How tested:
    - Established a test release pipeline for this, builds created and pushed to registry successfully: https://dlwteam.visualstudio.com/Next/_releaseProgress?_a=release-environment-logs&releaseId=3161&environmentId=20910.
    - See packages connected to this repo in: https://github.com/microsoft/HydraLab/pkgs/container/hydra-lab-uber.
  - **Run with the docker image:**
    - **docker pull ghcr.io/microsoft/hydra-lab-uber:latest**
    - **docker run -p 9886:9886 -e BLOB_CONNECTION_STR=${BLOB_CONNECTION_STR} ghcr.io/microsoft/hydra-lab-uber:latest**
- Bug fix:
  - Store default team for user init in avoid of breaking when running system with no-auth structure;
  - Use cache for DeviceGroupRelation entity to avoid sqlite lock;
  - Update scanning rule of Linux path to involve dashes in adb file path;

### CODE REVIEW

File: azure-pipelines-ci.yml [hunk 1]
```diff
@@ -119,6 +119,13 @@ stages:
         Contents: '*.jar'
         TargetFolder: '$(Build.ArtifactStagingDirectory)/agent_deploy'
       condition: and(succeeded(), eq(variables.fullBuild, 'true'))
+    - task: CopyFiles@2
+      displayName: Copy deploy uber files
+      inputs:
+        SourceFolder: 'center/uber_trial/'
```
Comment (hydraxman) [inline azure-pipelines-ci.yml:125]: What's the meaning of 'uber_trial'?

File: center/src/main/resources/application-docker.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        format_sql: false
+        order_inserts: true
+        order_updates: true
+        jdbc:
+          batch_size: 400
+          batch_versioned_data: true
+  security:
+    oauth2:
+      enabled: false
+
```
Comment (hydraxman) [inline center/src/main/resources/application-docker.yml:24]: Are all the above lines necessary?

File: azure-pipelines-ci.yml (hunk 1 above)
Comment (olivershen-wow) [reply azure-pipelines-ci.yml:125]: Just name for the docker's target, or maybe rename it to uber_out_of_box?

Comment (hydraxman) [reply azure-pipelines-ci.yml:125]: Could it be just 'uber'?

File: center/src/main/java/com/microsoft/hydralab/center/config/SecurityConfiguration.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
 @Configuration
 @EnableWebSecurity
 @EnableGlobalMethodSecurity(prePostEnabled = true)
-@Profile({"default", "dev", "release"})
+@Profile({"default", "dev", "release", "docker"})
```
Comment (hydraxman) [inline center/src/main/java/com/microsoft/hydralab/center/config/SecurityConfiguration.java:22]: Could we remove this line?

File: center/uber_trial/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    echo y | android update sdk --no-ui --all --filter platform-tools --force && \
+    apt-get install -y curl
+
+COPY ${CENTER_JAR_FILE} ${CENTER_JAR_FILE}
+COPY ${AGENT_JAR_FILE} ${AGENT_JAR_FILE}
+COPY ${STARTUP_FOLDER_PATH}/${CENTER_INIT_SQLITE} /hydra/data/${CENTER_INIT_SQLITE}
+COPY ${STARTUP_FOLDER_PATH}/start.sh start.sh
+
+# Connect package with repo HydraLab
+LABEL org.opencontainers.image.source=https://github.com/microsoft/HydraLab
```
Comment (hydraxman) [inline center/uber_trial/Dockerfile:39]: Could you attach the wiki link of this 'org.opencontainers.image.source' here as a comment?

File: center/src/main/resources/application-docker.yml (hunk 2 above)
Comment (olivershen-wow) [reply center/src/main/resources/application-docker.yml:4]: Resolved

File: center/src/main/java/com/microsoft/hydralab/center/config/SecurityConfiguration.java (hunk 3 above)
Comment (olivershen-wow) [reply center/src/main/java/com/microsoft/hydralab/center/config/SecurityConfiguration.java:22]: Without enabling this, InMemoryUserDetailsManager would be re-enabled, and some other issues would occur such as websocket cannot connect.

File: center/uber_trial/Dockerfile (hunk 4 above)
Comment (olivershen-wow) [reply center/uber_trial/Dockerfile:39]: Attached

File: azure-pipelines-ci.yml (hunk 1 above)
Comment (olivershen-wow) [reply azure-pipelines-ci.yml:125]: Updated

**Review by hydraxman — APPROVED** [review]

**Comment by hydraxman** [comment]
> * Add a out-of-box docker for center-agent assembly
>   
>   * Updated config ymls of docker version for agent/center each.
>   * Passed agent variables during docker build.
>   * Blob connection string should be passed to docker run after pulling the image.
>   * How tested:
>     
>     * Established a test release pipeline for this, builds created and pushed to registry successfully: https://dlwteam.visualstudio.com/Next/_releaseProgress?_a=release-environment-logs&releaseId=3161&environmentId=20910.
>     * See packages connected to this repo in: https://github.com/microsoft/HydraLab/pkgs/container/hydra-lab-uber.
>   * **Run with the docker image:**
>     
>     * **docker pull ghcr.io/microsoft/hydra-lab-uber:latest**
>     * **docker run -p 9886:9886 -e BLOB_CONNECTION_STR=${BLOB_CONNECTION_STR} ghcr.io/microsoft/hydra-lab-uber:latest**
> * Bug fix:
>   
>   * Store default team for user init in avoid of breaking when running system with no-auth structure;
>   * Use cache for DeviceGroupRelation entity to avoid sqlite lock;
>   * Update scanning rule of Linux path to involve dashes in adb file path;

Nice work, could you also update the README.md and add a subsection to provide the quick run guide with Docker? You may do this in a separate PR.

**Pull request merged**